src/array/groupBy.ts

```diff
@@ -30,18 +30,18 @@
  * //   ]
  * // }
  */
-export function groupBy<T, K extends PropertyKey>(arr: readonly T[], getKeyFromItem: (item: T) => K): Record<K, T[]> {
-  const result = {} as Record<K, T[]>;
+export function groupBy<T, K extends PropertyKey>(arr: readonly T[], getKeyFromItem: (item: T) => K): Record<K, [T, ...T[]]> {
+  const result = {} as Record<K, [T, ...T[]]>;
 
   for (let i = 0; i < arr.length; i++) {
     const item = arr[i];
     const key = getKeyFromItem(item);
 
     if (!Object.hasOwn(result, key)) {
-      result[key] = [];
+      result[key] = [item];
+    } else {
+      result[key].push(item);
```

Comment:
I figured it wouldn't hurt to update the implementation to match the soundness of the return type